src/main/java/seedu/duke/parser/Parser.java

```diff
@@ -0,0 +1,174 @@
+                lineSpaced[1] = lineSpaced[1].concat(" " + lineSpaced[i]);
+            }
+            fullDescription = lineSpaced[1].trim();
+        }
+        CommandType type;
+        switch (lineSpaced[0]) {
+        case "list":
+            type = CommandType.LIST;
+            break;
+        case "add":
+            type = CommandType.ADD;
+            break;
+        case "view":
+            type = CommandType.VIEW;
+            break;
+        case "find":
+            type = CommandType.FIND;
+            break;
+        case "editstep":
+            type = CommandType.EDITSTEP;
+            break;
+        case "editingredient":
+            type = CommandType.EDITINGREDIENT;
+            break;
+        case "delete":
+            type = CommandType.DELETE;
+            break;
+        case "help":
+            type = CommandType.HELP;
+            break;
+        case "clear":
+            type = CommandType.CLEAR;
+            break;
+        case "exit":
+            type = CommandType.EXIT;
+            break;
+        default:
+            type = CommandType.UNKNOWN;
+        }
+        return new Command(type, fullDescription);
+    }
+
+
+    private static boolean matchString(String input, String regex) {
+        String matcher = input;
+        int count = 0;
+        while (matcher.contains(regex)){
+            matcher=matcher.substring(matcher.indexOf(regex)+1);
+            ++count;
+        }
+        boolean isMatch = (count==1);
+        return isMatch;
+    }
+    /**
+     * Returns an Array of Strings containing the parsed full description
+     * of a <code>Recipe</code> into its name, ingredients and tag.
+     *
+     * @param description the full description of the <code>Recipe</code>.
+     * @return parsed ArrayList of Strings containing the recipe's name, ingredients list and tag.
+     * @throws IncompleteInputException if full description input is missing the description or due date or both.
+     */
+    public static ArrayList<String> parseRecipe(String description) throws IncompleteInputException {
+        ArrayList<String> parsed = new ArrayList<>();
+        if(!matchString(description,"n/") || !matchString(description,"i/")
+                || !matchString(description,"t/") || !matchString(description,"s/")){
+            throw new IncompleteInputException(RECIPE_WRONG_NAME_INGREDIENTS_TAG_STEP);
+        }
+        if(!description.substring(0,2).equals("n/")){
+            throw new IncompleteInputException(RECIPE_WRONG_LEADING_STRING);
+        }
+        String[] parsedName = description.split(" i/");
+        String[] parsedIngredientsTag = parsedName[1].split(" t/");
+        String[] parsedTagStep = parsedIngredientsTag[1].split(" s/");
+        parsed.add(parsedName[0].substring(2).trim());
+        parsed.add(parsedIngredientsTag[0].trim());
+        parsed.add(parsedTagStep[0].trim());
+        try {
+            parsed.add(parsedTagStep[1].trim());
+        }catch (Exception e){
```

Comment:
I'm not sure how the checker did not pick up on this, but there a code-style violation here. Do check your code thoroughly for these issues :)